### PR TITLE
update webpack to @^3.7.0

### PR DESCRIPTION
**Pull request by adispring**

### CODE REVIEW

File: libify.js [hunk 1]
```diff
@@ -31,12 +31,12 @@ function replacement(filepath, content, basedir, options) {
   const parts = content.match(jsTokens);
   const webpackConfigPath = path.relative(
     path.dirname(filepath),
-    options.webpackConfigPath || path.join(basedir, 'webpack.config')
+    options.webpackConfigPath || path.join(basedir, 'webpack.config'),
```
Comment (wenbing) [inline libify.js]: node v6 不支持

File: package.json [hunk 2]
```diff
@@ -42,18 +42,18 @@
     "send-data": "^8.0.0",
     "st": "^1.2.0",
     "tape": "^4.6.3",
-    "webpack": "^2.2.1",
+    "webpack": "^3.7.0",
```
Comment (wenbing) [inline package.json]: ^3 ?

File: package.json [hunk 3]
```diff
@@ -42,18 +42,18 @@
...
     "webpack-dev-server": "^2.3.0",
     "xhr-request": "^1.0.1"
   },
   "peerDependencies": {
-    "webpack": "^2.2",
-    "webpack-dev-server": "^2.3"
+    "webpack": "^3.7.0",
+    "webpack-dev-server": "^2.3.0"
```
Comment (wenbing) [inline package.json]: webpack-dev-server 需要更新么

File: package.json [hunk 4]
```diff
@@ -68,7 +68,7 @@
     "css-loader-bbq": "^1.0.3",
     "defined": "^1.0.0",
     "expose-loader": "^0.7.3",
-    "extract-text-webpack-plugin": "^2.0.0",
+    "extract-text-webpack-plugin": "^3.0.0",
```
Comment (wenbing) [inline package.json]: 主要变更在这里备注一下？

File: package.json [hunk 5]
```diff
@@ -42,18 +42,18 @@
...
-    "webpack": "^2.2",
-    "webpack-dev-server": "^2.3"
+    "webpack": "^3.7.0",
+    "webpack-dev-server": "^2.3.0"
   },
   "dependencies": {
     "autoprefixer": "^6.7.3",
     "babel-core": "^6.23.1",
-    "babel-loader": "^6.3.2",
+    "babel-loader": "^7.0.0",
```
Comment (wenbing) [inline package.json]: 这个需要一起升级么

File: package.json (hunk 3 above)
Comment (adispring) [reply package.json]: webpack-dev-server 不需要更新

File: package.json (hunk 4 above)
Comment (adispring) [reply package.json]: "extract-text-webpack-plugin": "3.0.0",

Bug Fixes
add missing options.ignoreOrder details in Error message (#539) (dd43832)

fix: css generation order issue see: webpack/webpack#5225

Code Refactoring
Apply webpack-defaults & webpack 3.x support (#540) (7ae32d9)

refactor: DeprecationWarning: Chunk.modules 543

BREAKING CHANGES
Enforces engines of "node": ">=4.3.0 < 5.0.0 || >= 5.10

Updates to Chunk.mapModules. This release is not backwards compatible with Webpack 2.x due to breaking changes in webpack/webpack#4764

Enforces peerDependencies of "webpack": "^3.1.0".

**Comment by wenbing** [comment]
DllReferencePlugin.js 不需要了，可以删除掉

**Pull request merged**